server-common/src/main/java/io/a2a/server/interceptors/AttributeExtractor.java

```diff
@@ -0,0 +1,8 @@
+package io.a2a.server.interceptors;
+
+import java.util.Map;
+
+@FunctionalInterface
+public interface AttributeExtractor {
```

Comment:
There is also https://opentelemetry.io/docs/zero-code/java/agent/annotations/#adding-attributes-to-the-span-with-spanattribute